### PR TITLE
[Fix](debug) Support dump const nullable Column from Block

### PR DESCRIPTION
**Pull request by zclllyybb**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: https://github.com/apache/doris/pull/41175

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [x] Other reason
          - it's just debug util.

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/core/block.cpp [hunk 1]
```diff
@@ -511,16 +511,13 @@ std::string Block::dump_data(size_t begin, size_t row_limit, bool allow_null_mis
...
                 if (data[i].type->is_nullable() && !data[i].column->is_nullable()) {
-                    if (is_column_const(*data[i].column)) {
-                        s = data[i].to_string(0);
-                    } else {
-                        assert(allow_null_mismatch);
-                        s = assert_cast<const DataTypeNullable*>(data[i].type.get())
-                                    ->get_nested_type()
-                                    ->to_string(*data[i].column, row_num);
-                    }
+                    assert(allow_null_mismatch);
```
Comment (zhangstar333) [inline be/src/vec/core/block.cpp:514]: type:                            column:
nullable_string ----->const(NULLABLE(String))
string -----> const(NULLABLE(String))
string----> NULLABLE(STRING)

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Review by zhangstar333 — APPROVED** [review]

**Review by Mryange — APPROVED** [review]

**Pull request merged**